src/contentScripts/other/selma/layout.ts

```diff
@@ -300,6 +303,24 @@ const NULL_TABLE: Graphing.GradeStat[] = [1, 1.3, 1.7, 2, 2.3, 2.7, 3, 3.3, 3.7,
   count: 1
 }))
 
+type Grade = {
+  grade: number
+  credits: number
+}
+
+function calculateGradePrediction(credits: number, grades: Grade[]): string {
+  let output = ''
+  let sum = 0
+  for (const grade of grades) {
+    sum += grade.credits * grade.grade
+  }
+  output = String((sum / credits).toPrecision(2))
+  if (output === 'NaN') {
```

Comment:
Can only be `NaN` if both `sum` and `credits` are `0`. I would suggest to check that before line 317 and return early of that is the case.